src/pkgcore/ebuild/repo_objs.py

```diff
@@ -170,6 +171,217 @@ def __init__(self, metadata_xml, manifest):
+    :ivar email: email address.
+    :type name: C{unicode} object or C{None}
+    :ivar name: full name
+    :type url: C{unicode} object or C{None}
+    :ivar url: project website URI
+    :type description: C{unicode} object or C{None}
+    :ivar description: full project description.
+    :type members: C{tuple} of C{ProjectMember}
+    :ivar members: project members
+    :type subprojects: C{tuple} of C{Subprojects}
+    :ivar subprojects: subprojects
+    """
+
+    __slots__ = ('email', 'name', 'url', 'description', 'members',
+                 'subprojects')
+
+    def __init__(self, email, name=None, url=None, description=None,
+                 members=(), subprojects=()):
+        if email is None:
+            raise ValueError('email for project must not be null')
+        self.email = email
+        self.name = name
+        self.url = url
+        self.description = description
+        self.members = members
+        self.subprojects = subprojects
+
+    def __str__(self):
+        if self.name is not None:
+            res = f'{self.name} <{self.email}>'
+        else:
+            res = self.email
+        if self.url is not None:
+            return f'{res} ({self.url})'
+        return res
+
+    @property
+    def leads(self):
+        """Project lead(s), if any."""
+        return tuple(m for m in self.members if m.is_lead)
+
+    @property
+    def recursive_members(self):
+        """All project members, including members inherited from subprojects."""
+        # collect all subprojects from which to inherit, recursively
+        subprojects = list(sp for sp in self.subprojects if sp.inherit_members)
+        subproject_emails = set(sp.email for sp in subprojects)
+        i = 0
+        while i < len(subprojects):
+            for sp in subprojects[i].subprojects:
+                if sp.inherit_members and sp.email not in subproject_emails:
+                    subprojects.append(sp)
+                    subproject_emails.add(sp.email)
+            i += 1
+
+        members = list(self.members)
+        emails = set(m.email for m in self.members)
+        for sp in subprojects:
+            if sp.inherit_members:
+                for m in sp.members:
+                    # avoid duplicating people
+                    if m.email not in emails:
+                        # drop lead bit
+                        m = ProjectMember(email=m.email,
+                                          name=m.name,
+                                          role=m.role)
+                        members.append(m)
+                        emails.add(m.email)
+        return tuple(members)
+
+
+class ProjectsXml(object):
+    """projects.xml parsed results
+
+    Attributes are set to -1 if unloaded, None if no entry, or the value
+    if loaded.
+    """
+
+    __slots__ = (
+        "__weakref__", "_projects", "_source"
+    )
+
+    def __init__(self, source):
+        self._source = source
+
+    @property
```

Comment:
Probably another jitted attr here since that's the behavior you're approximating.

Reply:
Done. Though I'm not sure if it actually increases readability here.